### PR TITLE
Core - Inputs Commitment Calculation

### LINKED ISSUE #362: Core - Inputs Commitment calculation
## Description

Inputs Commitment is a BLAKE2b-256 hash of the serialized outputs witch it is not calculated when creating a transaction.
The flow of creating an essence input is
* get output IDs from an address via `get_outputs_id()`
* fetch output metadata via `get_output` and add outputs into essence input list via `tx_essence_add_input`

the `tx_essence_add_input` only stores ID, Index, keypair in the list, a way to fix this issue is to keep the whole output data in the essence input list.

### PR DESCRIPTION
**Pull request by matjazv**
# Description of change

- Fixed #362 
- Fixed native tokens print function `(native_tokens_print)`
- Removed blank line in `pbkdf2_vectors.py` file

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

All unit tests passed.
`test_send_msg_tx_basic` successfully sent a message to the network (`ledgerInclusionState: "included"`).

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests using CTest that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/crypto/iota_crypto.h [hunk 1]
```diff
@@ -8,6 +8,38 @@
 #include <stdint.h>
 #include <stdlib.h>
 
+#ifdef CRYPTO_USE_SODIUM
```
Comment (oopsmonk) [inline src/crypto/iota_crypto.h]: the intention of iota_crypto is to provide an interface between 3rd party crypto libs and iota client lib. in this way, the client lib can integrate with different crypto libs as needed. exporting these crypto headers breaks this advantage which means 3rd party crypto libs compatibility needs to be considered in Core/Client modules, like [here](https://github.com/iotaledger/iota.c/pull/363/files#diff-4f022eaaaadd51600447cc4544bb7942cac5aae3fd479477cfa4e9daecf8da4cR74-R78)

```
if defined(CRYPTO_USE_SODIUM)
  crypto_generichash_blake2b_state blake_state;
#elif defined(CRYPTO_USE_BLAKE2B_REF)
  blake2b_state blake_state;
#endif
```

File: tests/crypto/pbkdf2_vectors.py [hunk 2]
```diff
@@ -14,7 +14,6 @@
 
 
 def print_vector(pwd, salt, iteration, dkLen):
-
```
Comment (oopsmonk) [inline tests/crypto/pbkdf2_vectors.py:17]: is it about code format?

Comment (oopsmonk) [inline tests/crypto/pbkdf2_vectors.py:17]: is it about code format?

**Review by oopsmonk — CHANGES_REQUESTED** [review]

File: src/core/models/inputs/utxo_input.h [hunk 3]
```diff
@@ -26,6 +27,7 @@ typedef struct {
   byte_t tx_id[IOTA_TRANSACTION_ID_BYTES];  ///< The BLAKE2b-256 hash of the transaction payload containing the
                                             ///< referenced output.
   uint16_t output_index;                    ///< The output index of the referenced output.
+  utxo_output_t *output;                    ///< Unspent output object. It is needed for calculating inputs commitment.
```
Comment (oopsmonk) [inline src/core/models/inputs/utxo_input.h]:
```suggestion

```

basically, the input list is selected outputs which we can just use `utxo_outputs_list_t` as input in tx essence?

```
typedef struct {
  ...
  utxo_outputs_list_t* inputs;
  byte_t inputs_commitment[CRYPTO_BLAKE2B_HASH_BYTES];
  utxo_outputs_list_t* outputs;
  ...
} transaction_essence_t;
```

Comment (oopsmonk) [reply src/core/models/inputs/utxo_input.h]: never mind, output response is not the case. :P

File: tests/crypto/pbkdf2_vectors.py (hunk 2 above)
Comment (matjazv) [reply tests/crypto/pbkdf2_vectors.py:17]: Yes. Code format tool removed a blank line.

File: src/crypto/iota_crypto.h (hunk 1 above)
Comment (matjazv) [reply src/crypto/iota_crypto.h]: Thanks for the explanation. Crypto headers will be moved back to their previous place and the code will be refactored.

**Review by oopsmonk — APPROVED** [review]

**Review by rebinmannady93 — APPROVED** [review]

**Pull request merged**